BAT/BCAux.h

```diff
@@ -36,6 +36,36 @@ class TH2;
 namespace BCAux
 {
 
+/**
+ * A trash to keep heap-allocated objects of type T alive until the
+ * trash goes out of scope. Then they are deleted. Ownership is not
+ * transferred in copies, only during a swap. */
+template <typename T>
+class BCTrash
+{
+public:
+    BCTrash() {}
+    BCTrash(const BCTrash<T>&) {}
+    BCTrash<T>& operator=(const BCTrash<T>&)
+    {
+        return *this;
```

Comment:
What is the meaning of this?

Reply:
As it says in the docstring, copy and assignment don't do anything. But they are implemented following the general interface so `BCTrash` plays well as a member of another class with compiler-generated copy ctor etc.